### PR TITLE
fix: Allow datatype update expressions for constructors with nameonly parameters

### LINKED ISSUE #1909: Datatype update syntax doesn't work with nameonly fields
```dafny
datatype ABC = ABC(nameonly a: int, nameonly b: int, nameonly c: int)

function JustChangeB(abc: ABC): ABC {
  abc.(b := 42) // Error: nameonly parameter 'a' must be passed using a name binding; it cannot be passed positionally
                // (same error repeated for every nameonly field)
}
```

### PR DESCRIPTION
**Pull request by RustanLeino**
Fixes #1909

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/Dafny/Resolver.cs [hunk 1]
```diff
@@ -15601,16 +15601,23 @@ public class ResolveOpts {
...
           } else {
-            ctor_args.Add(new ExprDotName(tok, d, f.Name, null));
+            ctorArg = new ExprDotName(tok, d, f.Name, null);
           }
+          ctorArguments.Add(ctorArg);
+          var bindingName = new Token(tok.line, tok.col) {
+            filename = tok.filename,
+            val = f.Name
+          };
+          actualBindings.Add(new ActualBinding(bindingName, ctorArg));
```
Comment (MikaelMayer) [inline Source/Dafny/Resolver.cs:15618]: Everything else is just refactoring, replacing `null` by `bindingName` is the actual fix, right?

**Review by MikaelMayer — APPROVED** [review]
Thanks for also cleaning up the code, along with fixing that issue.

File: Source/Dafny/Resolver.cs (hunk 1 above)
Comment (RustanLeino) [reply Source/Dafny/Resolver.cs:15618]: Correct.

**Review by MikaelMayer — APPROVED** [review]
Great job !

**Pull request merged**